### DISCUSSION TITLE
2022-03-09 Budget Subcomittee Meeting

### DISCUSSION THREAD

**Discussion by andy-blum** [ddev/ddev#3681] [DDEV Advisory Group]
## Agenda:

- Start recording
- Set next meeting (or recurring)
- Ask somebody to take notes on the attached doc
- Review everything that has happened! Lead sponsor, Github Sponsors, etc
- Discuss accountability for Github Sponsors money and an interim budget
- Plan next steps for group

## Minutes:
Recording started.

Next budget meeting:

- Start meeting every two months?
- Let&#39;s go with every other month, one week after the regular meeting, at 11A EST.

Randy conveyed his thanks for being here at &quot;such a boring meeting.&quot; :)

Review of what has happened:

- Blockchain-based ideas have gone away
- Someone pestered Randy to set up GitHub Sponsors on his personal account on Twitter, so he did that. Hadn&#39;t occurred to him to do that, so he did and pointed it at the bank account of the LocalDev Foundation. Sponsors info on GitHub does reflect that.
- There was a little &quot;drama&quot; with Fruition which is now in the past. But, they heard that Platform.sh heard that they would be willing to be a lead sponsor. Eventually Fruition did a deal and the trademark is now in the custody of Platform.sh. They wanted to sponsor as by mechanism of making Randy an employee, pay him a salary to maintain DDEV. This began Monday, March 7th and he became an employee of Platform.sh. The uncertainty about the trademark is resolved; the uncertainty around Github Organization is resolved.
  - Andy Blum - as far as the salary requirement you were looking for from DDEV, has Platform done at least that well? Yes, they did better + benefits; and he has better work time flexibility. Can be gone for the whole month of April, ex.g (Spain). Still not sure how I&#39;ll fit in with some of their more traditional roles/stuff, but they understand that this is what&#39;s important - still possibly TBD.
  - Second question AB - does Fruition have any ownership left at this point? Major stakeholder, etc? Randy - they are completely in the rearview now. At no point did they contribute anything that we then owe them. Platform&#39;s goals of the project are written into the contract, ex.g. That if Platform goes poof, then there&#39;s Good Faith written in to make sure it doesn&#39;t disrupt the project.
  - Chris Wells - what \_was\_ the uncertainty around the GitHub organization etc? Fruition didn&#39;t want to share org, ex.g. - but Platform does. So he has an &quot;Owner&quot; role on the GitHub Organization. Trademark? The old company DRUD had &quot;DDEV&quot; as a trademark. That was one of the assets Fruition bought, and it might&#39;ve ended up requiring a fork of the project. With Platform, Randy has a lot more confidence with Platform&#39;s willingness on some of these things.
  - Cary - so Platform.sh has these assets. DDEV is under Apache2 license. A fork and rename would&#39;ve been non-trivial so happy that the threat of that is gone. &quot;DRUD&quot; (the name) doesn&#39;t have a lot of good power to help people find the project.
  - Cary - so what is the role of the LocalDev foundation? Do we need it at all. Randy thinks it&#39;s fairly important to let people control how they donate money to support the project and this is an important part. Cary says Platform.sh should take ownership over some of the paperwork and legal requirements of what would be the LocalDev foundation.
    - Drud had GitHub Sponsors up a year or two ago (disabled it for some reason). It was odd because it was a commercial company saying &quot;you can send us some money&quot; and maybe it&#39;s also awkward for Platform.sh to do that--why should I send money to them? It&#39;s better to send money to the LocalDev foundation. Randy asks, &quot;what is the avenue for people who want to contribute?&quot; And argues that we do have that right now. But maybe now is the time to fiddle with that?
    - CG: Platform should be instrumental in setting up this entity due to its financial backing.
    - LocalDev foundation is a Colorado non-profit. It is not yet a 501(c)3, but it perhaps could be. Making it a 501(c)3 would be an admirable goal, but we need to know if the charter of the org is [compatible with a 501c3](https://www.irs.gov/charities-non-profits/charitable-organizations/exemption-requirements-501c3-organizations).
    - Randy doesn&#39;t want to do the work of setting up the 501(c)3, and Cary argues that Platform.sh should do that work. Maybe we revisit this in a couple months once the dust settles. And Randy agrees that Platform does have more resources to fight that fight.
    - CG argues that we keep Randy&#39;s focus on the work of maintenance of the project, and not the legwork of organization-creating.

Reviewing the state of affairs with the dollars:

- Was impressed with the GitHub Sponsors income. Monthly estimated income is $1665. First payout estimate (March 22nd) is $5800. (Does that buy us new test runners?)
  - That does buy us new test runners. I (RF) think that&#39;s the first order of business.
  - Would like to buy or rent a couple Mac M1 instances. [MacStadium](https://www.macstadium.com/) has them… RF has applied for the Open Source grant at MacStadium which would get him 1.
  - His Mac M1 runs tests now but he wants to bring it to Europe. :)
  - Do we support spending 1-2k of this incoming funds to rent test runners?
    - AB: how often does this expense occur? Need 5 over the next year but then that might be it for the next 5 years, unless the project starts to require more test cycles.
    - MacStadium is real Macs, so compatible with the Docker Desktop licensing, etc.
    - We have tried Azure but it has been unreliable.
    - Ideally we&#39;d like to move to cloud, but RF hasn&#39;t figured out how to do that.
    - One thing I do like about the physical machines is that everything is &quot;touchable&quot; BUT the downside of that is that it means ONLY Randy can do some of those things.
    - Do we need to replace everything now, or do we need to replace a few?
    - CG volunteered to at least look at Cloud stuff for Windows - RF points out that Docker Desktop cannot run on Windows Server.
    - There are about 15 physical machines (laptops) that run under BuildKite (and then there&#39;s also GitHub Actions and CircleCI expenses)
    - CG (in chat): Maybe someone should approach Docker about a license exception. Maybe me (CG)
  - RF proposal: spend 1-2k of the first payout on either Cloud or physical machine (probably physical Mac mini M1s and maybe an experimental Windows mini PC).
    - The oldest machines are the 2013 Macs.
    - The windows machines are also pretty old, but they are doing their job.
    - We may not need Docker Desktop w/ version 1.19 (except on traditional Windows), BUT the vast majority of people are using it, so we need to test it. But we can test Mac right on GitHub using Colima.
    - Andrew Berry (in chat): We could build a windows image with docker installed, My 2c is Mac stadium is what we should lean towards vs more physical machines
    - **Agreement: OK to spend 1-2k over the next month to replace up to three test runners, be it with MacStadium or physical machines.**
- Next budget meeting we will present financial statements.
- Want to spend all our money each year, otherwise the government takes it. We can also push the money &quot;upstream&quot; if needed. Randy&#39;s committed to 10% going upstream. PayPal is set up to be used for GitHub Sponsors - all this is PayPal linkage + debit card for the LocalDev account. This enables us to pay people upstream with GitHub Sponsors (or Ko-fi or whatever). (Randy has been pestering Mutagen folks to get GHSpon enabled so he can support them.) debsury also is a repo we support.
- Discord Nitro / boost - gives us the 3-day archiving on threads. May give us other things that we&#39;re not as interested in, but this is $10/mo.
  - Note that threads are all still there and can be unarchived just by speaking in it.

## Action Items

- [ ] RANDY FAY - take a tour of our testing resources now so we can explain them (YouTube Video?)
- [x] ANDY BLUM - open [github discussion to propose/discuss upstream project contributions](https://github.com/drud/ddev/discussions/3682)
- [X] ANDY BLUM - move this document to a GitHub discussion

## Recording

See [zoom link](https://www.dropbox.com/s/nogghk4khg5hpmt/20220309%20DDEV%20Budget%20Committee.mp4?dl=0)

**Comment by rfay** [comment]
Added zoom link.
